internal/action/command.go

```diff
@@ -538,60 +539,62 @@ func setGlobalOptionNative(option string, nativeValue interface{}, write bool) e
 	}
 
 	// ...if it's not local continue with the globals
-	config.GlobalSettings[option] = nativeValue
-	config.ModifiedSettings[option] = true
-	delete(config.VolatileSettings, option)
-
-	if option == "colorscheme" {
-		// LoadSyntaxFiles()
-		config.InitColorscheme()
-		for _, b := range buffer.OpenBuffers {
-			b.UpdateRules()
-		}
-	} else if option == "infobar" || option == "keymenu" {
-		Tabs.Resize()
-	} else if option == "mouse" {
-		if !nativeValue.(bool) {
-			screen.Screen.DisableMouse()
-		} else {
-			screen.Screen.EnableMouse()
-		}
-	} else if option == "autosave" {
-		if nativeValue.(float64) > 0 {
-			config.SetAutoTime(nativeValue.(float64))
+	if !reflect.DeepEqual(config.GlobalSettings[option], nativeValue) {
```

Comment:
How about:
```
if reflect.DeepEqual(config.GlobalSettings[option], nativeValue) {
	if write {
		// Although the setting in micro hasn't changed, the setting on the disk
		// may have changed. Make sure to update it.
		return config.WriteSettings(filepath.Join(config.ConfigDir, "settings.json"))
	}
	return nil
}

...
```

to keep the rest of the function out of this huge `if`?

Reply:
> So why not just return `err` in this case?

It can be discussed, where the `screen.TermMessage(err)` is done and if `b.isModified = true` shall take place in this scenario or not. But I think you're more puzzled by the `b.Settings["encoding"] = "utf-8"` right?
The reason was to get informed again in the same life cycle in case the error in the `settings.json` wasn't resolved, but `reload` used again...processing only upon change.

Reply:
> The reason was to get informed again in the same life cycle in case the error in the `settings.json` wasn't resolved, but `reload` used again...processing only upon change.

Could  you decrypt what this means?

And whatever this means, as a side effect it affects not just `reload` but anyone using `SetOptionNative`, e.g. a plugin?

Reply:
> Could you decrypt what this means?

`settings.json`:
```json
{
    "encoding": "blub"
}
```

Action:
1. In `micro` perform multiple times `reload` with this line present in the `settings.json` with the above mentioned `b.Settings["encoding"] = "utf-8"`
2. In `micro` perform multiple times `reload` with this line present in the `settings.json` without the above mentioned `b.Settings["encoding"] = "utf-8"`

Result:
1. You're informed every time again, that the issue is still present in the file (at least this was the case before the change detection in `(s|S)etGlobalOptionNative()`)
2. You're informed just once, that the issue is present in the file

> And whatever this means, as a side effect it affects not just `reload` but anyone using `SetOptionNative`, e.g. a plugin?

Ok, I realize...the plugin itself isn't informed with the negative result (`err`).
We have talked past each other or at least I was to dumb to understand.

Since my arguments have vanished into air it's ok to follow your first advice here.

Reply:
Actually I was wrong with suggesting to validate `encoding` here. This is not the place where we validate options (especially since we actually already set `b.Settings[option] = nativeValue` before we get here). It should be validated in by option validators, it's their job.

And we actually already validate `encoding` in option validators, like other options. And the problem is that we don't always call option validators when we should. So if we add `"matchbracestyle": "aaa"` to `settings.json` and do `reload`, it will set `matchbracestyle` to `aaa`.